internal/app/referenceserver/impl.go

```diff
@@ -123,53 +134,56 @@ func (s *conformanceServer) ServerStream(
 	req *connect.Request[v1.ServerStreamRequest],
 	stream *connect.ServerStream[v1.ServerStreamResponse],
 ) error {
-	responseDefinition := req.Msg.ResponseDefinition
-	if responseDefinition != nil {
-		internal.AddHeaders(responseDefinition.ResponseHeaders, stream.ResponseHeader())
-		internal.AddHeaders(responseDefinition.ResponseTrailers, stream.ResponseTrailer())
-	}
-
 	// Convert the request to an Any so that it can be recorded in the payload
 	msgAsAny, err := asAny(req.Msg)
 	if err != nil {
 		return err
 	}
+
 	respNum := 0
-	for _, data := range responseDefinition.ResponseData {
-		resp := &v1.ServerStreamResponse{
-			Payload: &v1.ConformancePayload{
-				Data: data,
-			},
-		}
 
-		// Only set the request info if this is the first response being sent back
-		// because for server streams, nothing in the request info will change
-		// after the first response.
-		if respNum == 0 {
-			resp.Payload.RequestInfo = createRequestInfo(ctx, req.Header(), []*anypb.Any{msgAsAny})
-		}
+	responseDefinition := req.Msg.ResponseDefinition
+	if responseDefinition != nil { //nolint:nestif
+		internal.AddHeaders(responseDefinition.ResponseHeaders, stream.ResponseHeader())
+		internal.AddHeaders(responseDefinition.ResponseTrailers, stream.ResponseTrailer())
+
+		for _, data := range responseDefinition.ResponseData {
+			resp := &v1.ServerStreamResponse{
+				Payload: &v1.ConformancePayload{
+					Data: data,
+				},
+			}
 
-		time.Sleep((time.Duration(responseDefinition.ResponseDelayMs) * time.Millisecond))
+			// Only set the request info if this is the first response being sent back
+			// because for server streams, nothing in the request info will change
+			// after the first response.
+			if respNum == 0 {
+				resp.Payload.RequestInfo = createRequestInfo(ctx, req.Header(), []*anypb.Any{msgAsAny})
+			}
 
-		if err := stream.Send(resp); err != nil {
-			return connect.NewError(connect.CodeInternal, fmt.Errorf("error sending on stream: %w", err))
+			time.Sleep((time.Duration(responseDefinition.ResponseDelayMs) * time.Millisecond))
```

Comment:
nit: I didn't catch this in the original review, but this would be more readable if you factor out the sleep duration into a variable, above outside the loop.

Reply:
Sounds good. Should I do the same for unary and client streaming also? Feels like I should for consistency but they aren't in a loop and are sort of a one-shot deal. I'd probably be declaring the variable directly above the `Sleep` call.

Reply:
Sure, that would probably help readability in the other places, too.